edk2toollib/uefi/telem/parser_main.py

```diff
@@ -0,0 +1,660 @@
+        elif(self.SectionSeverity == 2):
+            return "Corrected"
+        elif(self.SectionSeverity == 3):
+            return "Informational"
+        
+        return ""
+
+    ##
+    # Parse out the custom string identifying the FRU hardware
+    ##
+    def FRUStringParse(self) -> str:
+
+        # Only parse if data is valid
+        if(self.ValidSectionBitsList[1]):
+
+            # Convert the FRU string from bytes to a string
+            try:
+                return self.FRUString.decode('utf-8')
+            except:
+                return "Unable to parse"
+        
+        return "Invalid"
+
+    ##
+    # Print relevant portions of the section header. Change to suit your needs
+    ##
+    def PrettyPrint(self) -> str:
+        string = ""
+        temp = ""
+
+        string = concat_n(string,concat("Section Length:    ",str(self.SectionLength)))
+
+        temp = self.FlagsParse()
+        string = concat_n(string,concat("Flags:             ",temp))
+
+        temp = self.SectionSeverityParse()
+        string = concat_n(string,concat("Section Severity:  ",temp))
+
+        temp = self.SectionTypeParse()
+        string = concat_n(string,concat("Section Type:      ",temp))
+
+        temp = self.FRUIDParse()
+        string = concat_n(string,concat("FRU ID:            ",temp))
+        
+        temp = self.FRUStringParse()
+        string = concat_n(string,concat("FRU String:        ",temp))
+        
+        return string
+
+##
+# Dumps byte code of input
+##
+def HexDump(input, bytesperline):
+
+    string      = "" # Stores the entire hexdump string
+    asc         = "" # Stores the ascii version of the current hexdump line
+    byte        = "" # Stores the base 16 byte version of the current hexdump line
+    rangecheck  = lambda x : x < 31 or x > 127 # Used to check if a byte value is within relevant ascii character bounds
+
+    # Go through every byte from the input
+    for i in range(len(input)):
+
+        # Add a byte string version of the byte onto the byte string
+        byte = concat_s(byte, format(input[i],'02X'))
+        
+        # Add an ascii version of the byte onto the ascii string
+        if rangecheck(input[i]):
+                asc = concat_s(asc,". ")
+        else:
+            asc = concat_s(asc,format(chr(input[i])," <2"))
+        
+        # Once we've reached bytesperline length, concatenate asc and byte strings and start a new line
+        if(not (i + 1) % bytesperline):
+            string = concat_n(string,concat(byte,asc))
+            asc = ""
+            byte = ""
+    
+    # Check if there are any remaining characters in the asc and byte strings to be added to string 
+    if(len(input) % bytesperline):  
+        string = concat(string,concat_n(concat(byte,"   " * (bytesperline - (len(byte)//3))),concat(asc,"   " * (bytesperline - (len(asc)//3)))))
+
+    return string
+
+##
+# Load friendly names from friendlynames.csv to FriendlyNames dict
+##
+def ImportFriendlyNames():
```

Comment:
You don't have any guarantees that this CSV is going to be colocated with this code. You can use pkg_resources but it gets weird fast